### PR TITLE
Add opentelemetry-python ASGI integration

### PR DESCRIPTION
**Pull request by florimondmanca**
**Checklist**

- [x] This project is explicitly related to ASGI.
- [x] The new list entry contains a project name, URL and description.

**What is this project?**

https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/asgi/asgi.html

OpenTelemetry is an in-the-works CNCF project for standardizing collection of telemetry metrics (via _instrumentations_) and their ingestion (via _exporters_) in a vendor-neutral way. https://opentelemetry.io/

It is supported by several vendors, such as Datadog, as well as other open source technologies (such as Prometheus).

`opentelemetry-python` ships with an ASGI middleware and some helpers to collect metrics from any ASGI apps.

**Do you know about other similar projects?**

No

**If so, how is this one different?**

/

**Context**
Found out about this while investingating https://github.com/encode/httpx/issues/1264 …

---

_Anyone who agrees with this pull request can add a 👍._

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,7 @@ _Packages for monitoring ASGI web applications._
 
 <!-- sort_by:name -->
 
+- [opentelemetry-python](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/asgi/asgi.html) - ASGI middleware and helpers for collecting application metrics via the (currently alpha) OpenTelemetry standard.
```
Comment (florimondmanca) [inline README.md:86]:
```suggestion
- [opentelemetry-python](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/asgi/asgi.html) - ASGI middleware and helpers for collecting application metrics via the (currently alpha) OpenTelemetry standard. Supports HTTP and WebSocket.
```

**Comment by florimondmanca** [comment]
cc @majorgreys - Happy to get your feedback on whether this addition would be relevant, since (according to git commits…) I believe you contributed the ASGI instrumentation for OpenTelemetry! 😃

**Comment by majorgreys** [comment]
Definitely relevant!

I would add there is also auto-instrumentation for Starlette and FastAPI (that draws on the ASGI middleware) in OpenTelemetry:

- https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/starlette/starlette.html
- https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/fastapi/fastapi.html

And while we are at it, also good to include that the Datadog Python APM client also provides an ASGI middleware for tracing. We are currently working on support for ASGI web frameworks as well and this repo would be a great place to publicize it when they are shipped.

- http://pypi.datadoghq.com/trace/docs/web_integrations.html#asgi

**Comment by florimondmanca** [comment]
> the Datadog Python APM client also provides an ASGI middleware for tracing

Yup great point. I thought I had added it already after the work done on migrating ddtrace-ASGI to dd-trace-py, but doesn't seem so!

> also auto-instrumentation for Starlette and FastAPI 

> support for ASGI web frameworks as well and this repo would be a great place to publicize it when they are shipped.

This list is exclusively focused on ASGI and I try to keep framework-specific entries at an absolute minimum, so most likely won't be keen to add entries or mentions for those framework specific integrations here. They'd very much have a place in awesome-fastapi or any such list, though!

File: README.md [hunk 2]
```diff
@@ -83,6 +83,7 @@ _Packages for monitoring ASGI web applications._
 
 <!-- sort_by:name -->
 
+- [opentelemetry-python](https://opentelemetry-python.readthedocs.io/en/stable/instrumentation/asgi/asgi.html) - ASGI middleware and helpers for collecting application metrics via the (currently alpha) OpenTelemetry standard. Supports HTTP and WebSocket.
```
Comment (florimondmanca) [inline README.md:86]:
```suggestion
- [opentelemetry-python](https://opentelemetry-python.readthedocs.io/en/latest/instrumentation/asgi/asgi.html) - ASGI middleware and helpers for collecting application metrics via the (currently alpha) OpenTelemetry standard. Supports HTTP and WebSocket.
```

**Pull request merged**